versioned_plugins.rb

```diff
@@ -429,6 +432,41 @@ def load_alias_definitions_for_target_plugins(plugin_names_by_type)
 
     aliases
   end
+
+  def fetch_stack_versions
+    stack_versions = Net::HTTP.get(URI.parse(VERSIONS_URL))
+    puts "Logstash version: #{get_logstash_version(stack_versions)}\n"
+    puts "ECS version: #{get_ecs_version(stack_versions)}\n"
+  end
+
+  def get_logstash_version(stack_versions)
+    version = stack_versions[/\:logstash_version: (.*?)\n/, 1]
+    @logstash_version = version.strip unless version.nil?
+  end
+
+  def get_ecs_version(stack_versions)
+    version = stack_versions[/\:ecs_version: (.*?)\n/, 1]
+    @ecs_version = version.strip unless version.nil?
+  end
+
+  def write_stack_versions(content, type)
+    # BRANCH and ECS_VERSION are newly added, will be available when every plugin index docs are re-indexed.
+    # This is a backfill logic to add the fields after :type: entry
+    match = content.match(/\[":branch: %BRANCH%"\]/)
+    if (match.nil?)
+      type_entry = ":type: #{type}\n"
+      logstash_version_entry = ":branch: %BRANCH%\n"
+      ecs_version_entry = ":ecs_version: %ECS_VERSION%\n"
+      index = content.index(type_entry)
+      index = index + type_entry.length
+      content.insert(index, logstash_version_entry)
+      content.insert(index + logstash_version_entry.length, ecs_version_entry)
+    end
```

Comment:
minor nitpick: since we're not doing anything with the resulting match object, we can use another way to match against a regex:

```
irb(main):006:0> puts "match" if "hey" =~ /h/
match
=> nil
irb(main):007:0> puts "match" if "hey" =~ /a/
=> nil
```

And in this case:

```suggestion
    if content =~ /[":branch: %BRANCH%"\]/
      type_entry = ":type: #{type}\n"
      logstash_version_entry = ":branch: %BRANCH%\n"
      ecs_version_entry = ":ecs_version: %ECS_VERSION%\n"
      index = content.index(type_entry)
      index = index + type_entry.length
      content.insert(index, logstash_version_entry)
      content.insert(index + logstash_version_entry.length, ecs_version_entry)
    end
```